src/Application/Messenger/Handler/MandateUpsertedHandler.php

```diff
@@ -87,6 +88,55 @@ public function __invoke(MandateUpserted $message): void
         }
     }
 
+    /**
+     * Try to safely set Salesforce ID, and other push tracking fields. If it
+     * fails repeatedly, this should be safe to leave for a later update.
+     * Salesforce has UUIDs so we won't lose the ability to reconcile the records.
+     *
+     * @param Salesforce18Id<RegularGivingMandate>|null $salesforceId
+     */
+    private function setSalesforceFieldsWithRetry(
+        MandateUpserted $changeMessage,
+        ?Salesforce18Id $salesforceId
+    ): void {
+        $tries = 0;
+        $uuid = $changeMessage->uuid;
+
+        do {
+            try {
+                if ($tries > 0) {
+                    /** @psalm-suppress InvalidCast There's a bug analysing do/while w.r.t. $tries */
+                    $this->logger->info("Retrying setting Salesforce fields for mandate $uuid after $tries tries");
+                }
+                $this->setSalesforceFields($uuid, $salesforceId);
+                return;
+            } catch (DBALException\RetryableException $exception) {
```

Comment:
yep makes sense to catch any RetryableException - I guess the message really only applies if its LockWaitTimeoutException rather than a DeadlockException but not really an issue.